File: app/src/main/java/org/wikipedia/feed/dayheader/DayHeaderMarginItemDecoration.java
package org.wikipedia.feed.dayheader;

import android.content.Context;
import android.graphics.Rect;
import android.view.View;

import androidx.annotation.DimenRes;
import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import org.wikipedia.views.MarginItemDecoration;

public class DayHeaderMarginItemDecoration extends MarginItemDecoration {

    public DayHeaderMarginItemDecoration(@NonNull Context context, @DimenRes int bottomId) {
        super(0, 0, 0, pixelSize(context, bottomId));
    }

    @Override public void getItemOffsets(Rect outRect, View view, RecyclerView parent,
                                         RecyclerView.State state) {
        if (view instanceof DayHeaderCardView) {
            super.getItemOffsets(outRect, view, parent, state);
        }
    }
}

File: app/src/main/java/org/wikipedia/feed/dayheader/DayHeaderMarginItemDecoration.kt
package org.wikipedia.feed.dayheader

import android.content.Context
import android.graphics.Rect
import android.view.View
import androidx.annotation.DimenRes
import androidx.recyclerview.widget.RecyclerView
import org.wikipedia.views.MarginItemDecoration

class DayHeaderMarginItemDecoration(context: Context, @DimenRes bottomId: Int) :
        MarginItemDecoration(0, 0, 0, context.resources.getDimensionPixelSize(bottomId)) {
    override fun getItemOffsets(outRect: Rect, view: View, parent: RecyclerView,
                                state: RecyclerView.State) {
        if (view is DayHeaderCardView) {
            super.getItemOffsets(outRect, view, parent, state)
        }
    }
}
